### PR TITLE
Use a newer version of dartdoc

### PR DESCRIPTION
**Pull request by sethladd**

### CODE REVIEW

**Comment by sethladd** [comment]
wdyt @abarth ?

closes https://github.com/flutter/flutter/issues/935

File: travis/test.sh [hunk 1]
```diff
@@ -21,6 +21,8 @@ flutter test --flutter-repo
 
 if [ $TRAVIS_PULL_REQUEST = "false" ]; then
   if [ $TRAVIS_BRANCH = "master" ]; then
+    pub global activate dartdoc 0.8.4
+    export PATH="$PATH":"~/.pub-cache/bin"
```
Comment (eseidelGoogle) [inline travis/test.sh]: I'm pretty certain this exists earlier. I definitely see it in the environment for travis.

I'm confused how we don't already have the latest version of dartdoc?

Comment (abarth) [inline travis/test.sh]: > I'm confused how we don't already have the latest version of dartdoc?

We use the version in the Dart SDK, which was frozen when the SDK was produced.  This command gives us the a newer version that was uploaded to pub.

Comment (abarth) [inline travis/test.sh]: Instead, of manipulating the path, please just run the `~/.pub-cache/bin/dartdoc` directly.

Also, you shouldn't ever use `~` in the PATH environment variable.  Not all consumers of PATH understand how to expand `~`.  The fact that the pub tool tells you to use this string is a bad recommendation.

Comment (sethladd) [inline travis/test.sh]: Thanks for the comments! I'll call dartdoc directly.

**Comment by sethladd** [comment]
OK, ptal.

**Comment by abarth** [comment]
LGTM

**Pull request merged**